lib/inspec/resources/sys_info.rb

```diff
@@ -3,117 +3,182 @@
-      end
+    def domain
+      @cache[:domain]
     end
 
-    def mac_hostname(opt = nil)
-      if !opt.nil?
-        opt = case opt
-              when "f", "long", "fqdn", "full"
-                " -f"
-              when "s", "short"
-                " -s"
-              else
-                "ERROR"
-              end
-      end
-      if opt == "ERROR"
-        skip_resource "The `sys_info.hostname` resource is not supported with that option on your OS."
-      else
-        inspec.command("hostname#{opt}").stdout.chomp
-      end
+    def ip_address
+      @cache[:ip]
     end
 
-    # returns the Manufacturer of the local system
     def manufacturer
-      os = inspec.os
-      if os.darwin?
-        "Apple Inc."
-      elsif os.linux?
-        inspec.command("cat /sys/class/dmi/id/sys_vendor").stdout.chomp
-      elsif os.windows?
-        inspec.powershell("Get-CimInstance -ClassName Win32_ComputerSystem | Select Manufacturer -ExpandProperty Manufacturer").stdout.chomp
-      else
-        skip_resource "The `sys_info.manufacturer` resource is not supported on your OS yet."
-      end
+      @cache[:manufacturer]
     end
 
-    # returns the ServerModel of the local system
     def model
-      os = inspec.os
-      if os.darwin?
-        inspec.command("sysctl -n hw.model").stdout.chomp
-      elsif os.linux?
-        inspec.command("cat /sys/class/dmi/id/product_name").stdout.chomp
-      elsif os.windows?
-        inspec.powershell("Get-CimInstance -ClassName Win32_ComputerSystem | Select Model -ExpandProperty Model").stdout.chomp
-      else
-        skip_resource "The `sys_info.model` resource is not supported on your OS yet."
+      @cache[:model]
+    end
+
+    def totalmemory_kb
+      @cache[:totalmemory_kb]
+    end
+
+    def hostname(opt = nil)
+      case opt
+        when nil
+          @cache[:hostname]
+        when "f","long","fqdn","full"
+          @cache[:fqdn]
+        when "d","domain"
+          @cache[:domain]
+        when "i", "ip_address"
+          @cache[:ip_address]
+        when "s","short"
+          @cache[:hostname]
+        else
+          skip_resource "The sys_info.hostname resource is not supported with that option."
       end
     end
 
+    def fqdn
+      @cache[:fqdn]
+    end
+
     def to_s
       "System Information"
     end
   end
+
+  class LinuxInfo < SysInfo
+    attr_reader :inspec
+    def initialize(inspec)
+      @inspec = inspec
+    end
+
+    def info
+      fqdn = inspec.command("hostname -f").stdout.chomp
+      domain = inspec.command("hostname -d").stdout.chomp
+      ip = inspec.command("hostname -I").stdout.chomp
+      short = inspec.command("hostname -s").stdout.chomp
+      manufacturer = inspec.command("cat /sys/class/dmi/id/sys_vendor").stdout.chomp
+      model = inspec.command("cat /sys/class/dmi/id/product_name").stdout.chomp
+      totalmemory_kb = inspec.command("awk '/MemTotal/ {print $2}' /proc/meminfo").stdout.chomp
```

Comment:
I would probably create these as methods in the SysInfo subclasses, we'll get lots of bonus usability.

```
linux = LinuxInfo.new(...)
foo.fqdn
foo.ip
```

Reply:
Here , i dont get your point. The resource have the methods within "sys_info" what is the advantage to have a copy of that foreach OS?
The subclasses foreach OS is only available inside sys_info ressource so its not usable directly inside the inspec tests.
Even if i change it in that way i would have to create a complex way of execution and caching for the windows side.
Or i have to split the 2 windows commands into 6 single powershell commands where i think this even dont bring any advantage.